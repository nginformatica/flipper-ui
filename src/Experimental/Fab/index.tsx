export { default as Fab } from './Fab'
export { default as FabWrapper } from './FabWrapper'

// export more components, hooks or helpers below if needed
