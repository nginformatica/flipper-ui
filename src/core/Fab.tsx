import { Fab as MuiFab } from '@material-ui/core'
import React, { MouseEvent } from 'react'
import { DefaultProps } from './types'

interface FabProps extends DefaultProps {
    disabled?: boolean
    component?: string
    color?: 'default' | 'primary' | 'inherit' | 'secondary'
    size?: 'small' | 'medium' | 'large'
    href?: string
    variant?: 'round' | 'extended'
    onClick?: (event?: MouseEvent<HTMLElement>) => void
    children: React.ReactElement<{}>
}

const Fab = ({
    children,
    margin,
    padding,
    style = {},
    variant,
    ...otherProps
}: FabProps) => (
    <MuiFab
        {...otherProps}
        variant={variant}
        style={{ margin, padding, ...style }}>
        {children}
    </MuiFab>
)

export default Fab
