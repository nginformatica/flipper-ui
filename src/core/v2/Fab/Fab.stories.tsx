import React from 'react'
import FabV2 from './Fab'
import FabWrapper from './FabWrapper'
import FileCopyIcon from '@material-ui/icons/FileCopy'

export const fab = () => (
    <FabV2 name='copy' tooltip='Copy file' onClick={() => alert('fab click')}>
        {<FileCopyIcon />}
    </FabV2>
)

export default {
    title: 'V2/Fab',
    component: FabV2,
    subcomponents: {
        FabWrapper
    }
}
