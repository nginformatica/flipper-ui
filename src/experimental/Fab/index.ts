export { default as Fab } from './Fab'
export { default as FabWrapper } from './FabWrapper'
