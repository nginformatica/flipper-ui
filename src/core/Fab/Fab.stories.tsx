import React from 'react'
import { Meta, StoryFn } from '@storybook/react'
import Fab from '.'

export default {
    title: 'Fab',
    component: Fab
} as Meta<typeof Fab>

const Template: StoryFn<typeof Fab> = args => <Fab {...args} />
export const Default = Template.bind({})

Default.args = {
    children: 'Fab'
}
