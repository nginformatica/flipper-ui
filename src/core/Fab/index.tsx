import { Fab as MuiFab } from '@material-ui/core'
import React, { MouseEvent } from 'react'
import { DefaultProps } from '../types'

export interface FabProps extends DefaultProps {
    disabled?: boolean
    component?: string
    color?: 'default' | 'primary' | 'inherit' | 'secondary'
    size?: 'small' | 'medium' | 'large'
    href?: string
    variant?: 'round' | 'extended'
    onClick?: (event?: MouseEvent<HTMLElement>) => void
    children: React.ReactElement<Record<string, unknown>>
}

const Fab = ({
    children,
    margin,
    padding,
    style = {},
    variant,
    ...otherProps
}: FabProps) => (
    <MuiFab
        {...otherProps}
        variant={variant}
        style={{ margin, padding, ...style }}>
        {children}
    </MuiFab>
)

export default Fab
