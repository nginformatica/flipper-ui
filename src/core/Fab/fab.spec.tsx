import * as React from 'react'
import { render, screen } from '@testing-library/react'
import Fab from '.'

describe('Fab', () => {
    it('should render', () => {
        render(
            <Fab data-testid='fab-container'>
                <>Fab</>
            </Fab>
        )

        const fab = screen.getByText('Fab')
        const container = screen.getByTestId('fab-container')

        expect(fab).toBeDefined()
        expect(container).toBeDefined()
    })

    it('should render with custom style', () => {
        render(
            <Fab
                data-testid='fab-container'
                margin={10}
                padding={5}
                style={{ backgroundColor: 'blue' }}>
                <>Fab</>
            </Fab>
        )

        const container = screen.getByTestId('fab-container')

        expect(container).toHaveProperty('style.margin', '10px')
        expect(container).toHaveProperty('style.padding', '5px')
        expect(container).toHaveProperty('style.backgroundColor', 'blue')
    })

    it('should match snapshot', () => {
        const { container } = render(
            <Fab data-testid='fab-container'>
                <>Fab</>
            </Fab>
        )
        expect(container).toMatchSnapshot()
    })
})
