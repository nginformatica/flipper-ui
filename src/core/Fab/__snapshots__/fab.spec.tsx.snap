// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fab should match snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiFab-root"
    data-testid="fab-container"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiFab-label"
    >
      Fab
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
